### PR TITLE
Fix LNOT

### PR DESCRIPTION
**Pull request by mmisono**
CreateUint() expects bit size, not byte size. This fixes the following
error.

```
% sudo ./src/bpftrace -e 'BEGIN { !(int32)0 }'
bpftrace: /home/ubuntu/work/bpftrace/src/types.cpp:251: bpftrace::SizedType bpftrace::CreateInteger(size_t, bool): Assertion `bits == 0 || bits == 8 || bits == 16 || bits == 32 || bits == 64' failed.
zsh: abort      sudo ./src/bpftrace -e 'BEGIN { !(int32)0 }'
```

##### Checklist

- [ ] Language changes are updated in `docs/reference_guide.md`
- [x] User-visible and non-trivial changes updated in `CHANGELOG.md`
- [x] The new behaviour is covered by tests

### CODE REVIEW

File: src/ast/semantic_analyser.cpp [hunk 1]
```diff
@@ -1594,7 +1594,19 @@ void SemanticAnalyser::visit(Unop &unop)
...
+          type.size == 4 || type.size == 8))
+    {
+      LOG(ERROR, unop.loc, err_)
+          << "The " << opstr(unop)
+          << " operator can not be used on expressions of type '" << type
+          << "'";
+    }
+    else
+    {
+      unop.type = CreateUInt(8 * type.size);
```
Comment (fbs) [inline src/ast/semantic_analyser.cpp:1608]: shouldn't we just return a fixed size int here?  Logical not should give a boolean so u64 might be better?

But not sure hwo much that will break.

Comment (mmisono) [reply src/ast/semantic_analyser.cpp:1608]: I'm not sure but with #1532, there is a chance that bytecode size becomes smaller?

Comment (fbs) [reply src/ast/semantic_analyser.cpp:1608]: Not sure. I Wouldn't mind  keeping this PR as is an investigate it further in #1532 & friends

**Comment by mmisono** [comment]
rebased

**Pull request merged**